presto-main/src/main/java/com/facebook/presto/operator/scalar/CharacterStringCasts.java

```diff
@@ -69,31 +71,30 @@ public static Slice varcharToCharCast(@LiteralParameter("y") Long y, @SqlType("v
     @LiteralParameters({"x", "y"})
     public static Slice charToVarcharCast(@LiteralParameter("y") Long y, @SqlType("char(x)") Slice slice)
     {
-        int textLength = countCodePoints(slice);
-        int resultLength = y.intValue();
+        return padSpacesAndTruncateToLength(slice, y.intValue());
```

Comment:
This doesn't seem right, shouldn't this pad to length `x` and then truncate to `y`? Otherwise `CAST(CHAR(2)'a' AS VARCHAR(3))` will result in two spaces at the end, instead of one